### ISSUE TITLE
Callbacks taking `NSError*` do not use the exact type

### ISSUE THREAD

**Issue by brianquinlan** [dart-lang/native#1504]
For the following methods:

```objc
// /System/Volumes/Data/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/System/Library/Frameworks/Foundation.framework/Versions/C/Headers/NSURLSession.h
- (void)sendMessage:(NSURLSessionWebSocketMessage *)message completionHandler:(void (NS_SWIFT_SENDABLE ^)(NSError * _Nullable error))completionHandler;

- (void)receiveMessageWithCompletionHandler:(void (NS_SWIFT_SENDABLE ^)(NSURLSessionWebSocketMessage * _Nullable message, NSError * _Nullable error))completionHandler;
```

This Dart code is generated:

```dart
  void sendMessage_completionHandler_(
      NSURLSessionWebSocketMessage message,
      objc.ObjCBlock<ffi.Void Function(ffi.Pointer<objc.ObjCObject>)>
          completionHandler)

  void receiveMessageWithCompletionHandler_(
      objc.ObjCBlock<
              ffi.Void Function(
                  ffi.Pointer<objc.ObjCObject>, ffi.Pointer<objc.ObjCObject>)>
          completionHandler)
```

Note how the `ObjCBlock` are called with a `ffi.Pointer<objc.ObjCObject>` rather than a `NSError?`.

**Comment by liamappelbe** [comment]
Might be fixed by one of the little fixes in #1491. I'll see if it repros after that

**Comment by liamappelbe** [comment]
Fixed by https://github.com/dart-lang/native/pull/1540
